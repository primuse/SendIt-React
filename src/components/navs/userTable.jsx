import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const tableHeading = (<tr id='table-head'>
  <th>ID</th>
  <th>First Name</th>
  <th>Last Name</th>
  <th>Admin</th>
  <th>Email</th>
  <th>Registered On</th>
</tr>);

const mapUsers = (allUsers) => {
  const userItems = allUsers.map((user, index) => (
    <tr key={index}>
      <td>SD0{user.id}</td>
      <td>{user.firstname}</td>
      <td>{user.lastname}</td>
      <td>{user.isadmin.toString()}</td>
      <td>{user.email}</td>
      <td>{user.registered}</td>
      <td>
        <Link to='/edit_user' className='btn xsm bg-bright-blue white'>
          Edit
          </Link>
      </td>
    </tr>
  ));
  return userItems;
};

export const UserTable = ({ allUsers }) => <div className='dash-cont'>
  <div id='table-info'>
    <a href='#'>All</a>
  </div>
  <div id='table-cont'>
    <table>
      <tbody>
        {tableHeading}
        {mapUsers(allUsers)}
      </tbody>
    </table>
  </div>
</div>;


UserTable.propTypes = {
  allUsers: PropTypes.array,
};
