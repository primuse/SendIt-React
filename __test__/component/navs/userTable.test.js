import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { UserTable } from '../../../src/components/navs/userTable.jsx';

const users = [
  {
    id: '1',
    firstName: 'Tiku',
    lastName: 'Okoye',
    isAdmin: 'false',
    email: 'cim@gmail.com',
    registered: '22-03-2019',
  },
  {
    id: '1',
    firstName: 'Tiku',
    lastName: 'Okoye',
    isAdmin: 'false',
    email: 'cim@gmail.com',
    registered: '22-03-2019',
  }
];

describe('IUser Parcel Table component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter>
        <UserTable allUsers={users} />
      </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
