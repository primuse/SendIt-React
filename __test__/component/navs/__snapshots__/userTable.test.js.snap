// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IUser Parcel Table component renders correctly 1`] = `
<div
  className="dash-cont"
>
  <div
    id="table-info"
  >
    <a
      href="#"
    >
      All
    </a>
  </div>
  <div
    id="table-cont"
  >
    <table>
      <tbody>
        <tr
          id="table-head"
        >
          <th>
            ID
          </th>
          <th>
            First Name
          </th>
          <th>
            Last Name
          </th>
          <th>
            Admin
          </th>
          <th>
            Email
          </th>
          <th>
            Registered On
          </th>
        </tr>
        <tr>
          <td>
            SD0
            1
          </td>
          <td />
          <td />
          <td />
          <td>
            cim@gmail.com
          </td>
          <td>
            22-03-2019
          </td>
          <td>
            <a
              className="btn xsm bg-bright-blue white"
              href="/edit_user"
              onClick={[Function]}
            >
              Edit
            </a>
          </td>
        </tr>
        <tr>
          <td>
            SD0
            1
          </td>
          <td />
          <td />
          <td />
          <td>
            cim@gmail.com
          </td>
          <td>
            22-03-2019
          </td>
          <td>
            <a
              className="btn xsm bg-bright-blue white"
              href="/edit_user"
              onClick={[Function]}
            >
              Edit
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
